packages/table/src/tableHotkeys.ts

```diff
@@ -52,8 +52,16 @@ export class TableHotkeys {
         this.props = props;
     }
 
-    public setState(state: TableState) {
-        this.state = state;
+    public setState(newState: TableState) {
+        if (
+            newState.focusedCell != null &&
+            (this.state.focusedCell == null || this.state.focusedCell !== newState.focusedCell)
+        ) {
+            // Ensure controlled focused cell is scrolled into view
+            this.scrollBodyToFocusedCell(newState.focusedCell);
```

Comment:
it looks like `focusedCell` can be provided in props which may cause this scroll call to be missed

```
// only set focused cell state if not specified in props
        if (this.props.focusedCell == null) {
            this.setState({ focusedCell });
        }
```

Reply:
thoughts on adding this `scrollBodyToFocusedCell` here while leaving it in the places it was already? is there no concern with calling this twice?

what are the cases this handled that are currently missed by `handleFocusMove` and `handleFocusMoveInternal` where this is called at the end?

Reply:
> what are the cases this handled that are currently missed by `handleFocusMove` and `handleFocusMoveInternal` where this is called at the end?

BLUF: It helps when the `focusedCell` is controlled (i.e. passed in via props) and we do not want to rely on native `Table2` navigation logic (`handleFocusMove`, `handleFocusMoveInternal`).

`handleFocusMove` and `handleFocusMoveInternal` correctly call `scrollBodyToFocusedCell`. But they do this even if `focusedCell` is controlled:

https://github.com/palantir/blueprint/blob/f2e9d6e049b3c52eef4ea7d9cb080b4dca23c7a1/packages/table/src/table2Utils.ts#L78-L85

Now, the table should not use its own navigation mechanism when `focusedCell` is controlled (I meant to make this a FLUP, in fact!).

Here's how this behavior affects us: we have to handle navigation during the capturing event phase and conditionally stop event propagation. We would therefore never hit `handleFocusMove` or `handleFocusMoveInternal`, meaning our controlled `focusedCell` would not get scrolled into view.

> it looks like focusedCell can be provided in props which may cause this scroll call to be missed

It should not, since we set the `focusedCell` on `Table2`'s state using props:

https://github.com/palantir/blueprint/blob/f2e9d6e049b3c52eef4ea7d9cb080b4dca23c7a1/packages/table/src/table2.tsx#L176-L181

> thoughts on adding this scrollBodyToFocusedCell here while leaving it in the places it was already? is there no concern with calling this twice?

I think it makes sense to remove the other two usages in `handleFocusMove` and `handleFocusMoveInternal`. Those two functions update the state, and we now plug this side-effect into the state change.